### PR TITLE
Dynamic content support

### PR DESCRIPTION
**Pull request by Ptr314**
Modifications for dynamic content support

### CODE REVIEW

File: zenpy/lib/mapping.py [hunk 1]
```diff
@@ -87,7 +87,8 @@ class ZendeskObjectMapping(object):
         'recipient_address': RecipientAddress,
         'recipient': Recipient,
         'response': Response,
-        'trigger': zenpy.lib.api_objects.Trigger
+        'trigger': zenpy.lib.api_objects.Trigger,
+        'dynamic_content_item': DynamicContent
```
Comment (facetoe) [inline zenpy/lib/mapping.py]: If you call this "item", you don't need to implement a request/response handler or a generator. It should all just work. I think it is better to follow the pattern of calling the object what Zendesk does where possible.

**Comment by facetoe** [comment]
Looks like the tests are failing as a property hasn't been implemented:

```
Traceback (most recent call last):
  File "/home/travis/build/facetoe/zenpy/tests/test_api/test_properties.py", line 17, in test_zendesk_object_properties_implemented
    self.check_properties_are_implemented(api_objects)
  File "/home/travis/build/facetoe/zenpy/tests/test_api/test_properties.py", line 37, in check_properties_are_implemented
    .format(object_name, missing_method_name))
AssertionError: zenpy.lib.api_objects.DynamicContent contains a property that calls a non existent method: '_get_default_locale'.
This method needs to be implemented in the zenpy.lib.api.Api class.
```

**Comment by Ptr314** [comment]
But the class was generated automatically. Can I solve it by just removing the wrong property? I don't undestand clearly why the script has decided to add it to the class.

**Comment by facetoe** [comment]
Ahh yea, it is a bit strange. I'll make a patch tomorrow and explain how it works in case you run into this again.

**Comment by Ptr314** [comment]
Removed the wrong property manually, also added an endpoint for targets.

**Comment by Ptr314** [comment]
Now added a possibility to edit brands & ticket forms

**Comment by facetoe** [comment]
@Ptr314 sorry for taking so long to review/merge this PR, I forgot all about it. I should have some time this weekend to have a look. Thanks!

**Comment by facetoe** [comment]
@Ptr314, I fixed up the property error. To explain how it works, many objects returned by the Zendesk API contain ids that link related objects. Zenpy resolves these lazily when requested, and it does this via properties on the Zenpy objects. For example, when you execute the following:

```python
ticket = zenpy_client.tickets(id=5396)
print(ticket.requester)
```

The `ticket.requester` access is actually a property: https://github.com/facetoe/zenpy/blob/master/zenpy/lib/api_objects/__init__.py#L3595

Which calls this method in `Api`: https://github.com/facetoe/zenpy/blob/master/zenpy/lib/api.py#L310

Now, depending on whether or not the User is cached, it will either retrieve it from the cache or execute an API call to Zendesk to retrieve it.

So the error you were seeing in the test is because the `_get_default_locale` was not yet implemented in the `Api` class. I added it in, and now it's all good: https://github.com/facetoe/zenpy/commit/2149baa7ccf4d82f3f65f590f791919575280456#diff-b8941e80c13585f2d92cacceaee1d6eeR402.

Thanks for the PR!

**Pull request merged**